packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -162,11 +165,30 @@ function typeDeclaredInPackage(
   );
 }
 
+function typeDeclaredInLib(
+  declarationFiles: ts.SourceFile[],
+  program: ts.Program,
+): boolean {
+  // Assertion: The type is not an error type.
+
+  // Intrinsic type (i.e. string, number, boolean, etc) - Treat it as if it's from lib.
+  if (declarationFiles.length === 0) {
+    return true;
+  }
+  return declarationFiles.some(declaration =>
+    program.isSourceFileDefaultLibrary(declaration),
+  );
+}
+
 export function typeMatchesSpecifier(
   type: ts.Type,
   specifier: TypeOrValueSpecifier,
   program: ts.Program,
 ): boolean {
+  // TODO: use `tsutils.isErrorType(type)` -- That function doesn't exists yet.
```

Comment:
Could you please add a test for error types as well?

Reply:
I added a test a little while back that should test this :)

Reply:
Thanks! :)